Dockerfile.dev

```diff
@@ -0,0 +1,22 @@
+FROM registry.fedoraproject.org/fedora:latest
+
+RUN mkdir /app
+WORKDIR /app
+
+# Some dependencies are only needed for aarch64
+RUN dnf install -y bubblewrap skopeo lvm2 rpm-ostree cryptsetup selinux-policy-targeted selinux-policy-devel qemu-img e2fsprogs \
+                    python3-devel python3-pyyaml python3-typer python3-attrs python3-tomli python3-pip \
+                    policycoreutils-python-utils python3-rich python3-jsonschema selinux-policy \
+                    make git findutils file coreutils curl qemu-img tar util-linux python-docutils pkg-config
+
+RUN dnf groupinstall -y rpm-development-tools
+
+RUN pip install pydevd_pycharm
+
+COPY . /app
+
+RUN make rpm-devel
+
+RUN rpm -i /app/rpmbuild/RPMS/*.rpm
+
+CMD [ "/usr/bin/osbuild", "--version" ]
```

Comment:
This is actually only a developer setup - so developers will use it and fix it if it does not work on their PC.
Not sure if this needs a CI, this is not used in releasing process or production/staging.

Reply:
Right, my concern is that if we do not make it part of the CI then it will break and we will not notice. This diminishes the usefulness of this feature because when a developer comes along (potentially weeks after it got broken) and wants to use it there is a risk that it's broken and this will be frustrating and trying to find the culprit will be harder (because potentially many unrelated changes have landed too). IMHO if this becomes part of the repo we should do some sort of (smoke) test to ensure it keeps working (unless of course the test is very costly?)

Reply:
It's not really too costly but would be something like an "downstream test" as this test only makes sense on the "osbuild-getting-started" level. Not sure if we want to integrate this here or make a recurring smoke-test in osbuild-getting-started (which would make sense once a week I guess)

Reply:
Thanks, your comment makes me wonder - if this test only makes sense on the "osbuild-getting-started" repo level, maybe the Dockerfile.dev should be moved there? (and just to be clear, I'm not passive aggressive here, this is a serious question). This way it's entirely up to the "getting-started" repo to implement this in whatever is the most practial way and "upstream" osbuild does not need to worry. 

I still think that even if it's just a "dev" setup it needs a (basic/smoke) test at the level that the "feature" is define. To me the analogy would be to have some code that is written in an upstream repo but only tested in a downstream unit test, that would feel unusual to me too.

P.S. Please do not resolve this (and some of the other conversations) quite so quickly. Mostly for the benefit of the other reviews as they often have very valuable input and when an issue is marked "resolved" this indicates that no further discussion is needed which seems to be not the case here.

Reply:
Defining build details and source dependencies in the osbuild-getting-started repo feels wrong. I'll add a smoke test here.

Reply:
I'm also not convinced this all belongs here.  We have the `.devcontainer` directory and that got out of date fast when the people that introduced it stopped using it.  I don't even know why we still keep it around, I assumed someone was using it but then we discovered it wasn't even working.

I'm not claiming everything we've ever added to this repository has been pristine and fully tested, but a broken `Dockerfile.dev` that's meant to help contributors but slowly became outdated is worse than not having one at all.  Having it in a separate repository for dev tooling seems more appropriate but even then I'd argue a guide explaining how things work so people can make their own workflows with a deeper understanding is more valuable.

Reply:
From my experience with the stack, it is too complicated to explain in a digestible document when "getting started". Until this is super easy and straight forward we need some mechanism to get things up and running, otherwise it will be always hard to get (good) first contributions, IMHO.

Reply:
I thought about this more and this would remove use cases where you might want to test the current source on composer or image-builder level with the same approach if not every repo can build a container _from source_
@achilleas-k @supakeen I'm available if you have time to test this yourself or we should do some session for pro/cons, just ping me